packages/ui/VirtualList/VirtualListBase.js

```diff
@@ -23,728 +27,771 @@ const gridListItemSizeShape = PropTypes.shape({
-
-		/**
-		 * Callback method of scrollTo.
-		 * Normally, `[Scrollable]{@link ui/Scrollable.Scrollable}` should set this value.
-		 *
-		 * @type {Function}
-		 * @private
-		 */
-		cbScrollTo: PropTypes.func,
-
-		/**
-		 * Client size of the list; valid values are an object that has `clientWidth` and `clientHeight`.
-		 *
-		 * @type {Object}
-		 * @property {Number} clientWidth - The client width of the list.
-		 * @property {Number} clientHeight - The client height of the list.
-		 * @public
-		 */
-		clientSize: PropTypes.shape({
-			clientWidth: PropTypes.number.isRequired,
-			clientHeight: PropTypes.number.isRequired
-		}),
-
-		/**
-		 * Data for passing it through `itemRenderer` prop.
-		 * NOTICE: For performance reason, changing this prop does NOT always cause the list to
-		 * redraw its items.
-		 *
-		 * @type {Any}
-		 * @default []
-		 * @public
-		 */
-		data: PropTypes.any,
-
-		/**
-		 * Size of the data.
-		 *
-		 * @type {Number}
-		 * @default 0
-		 * @public
-		 */
-		dataSize: PropTypes.number,
-
-		/**
-		 * Direction of the list.
-		 *
-		 * Valid values are:
-		 * * `'horizontal'`, and
-		 * * `'vertical'`.
-		 *
-		 * @type {String}
-		 * @default 'vertical'
-		 * @public
-		 */
-		direction: PropTypes.oneOf(['horizontal', 'vertical']),
-
-		/**
-		 * Called to get the props for list items.
-		 *
-		 * @type {Function}
-		 * @private
-		 */
-		getComponentProps: PropTypes.func,
-
-		/**
-		 * Number of spare DOM node.
-		 * `3` is good for the default value experimentally and
-		 * this value is highly recommended not to be changed by developers.
-		 *
-		 * @type {Number}
-		 * @default 3
-		 * @private
-		 */
-		overhang: PropTypes.number,
-
-		/**
-		 * It scrolls by page when `true`, by item when `false`.
-		 *
-		 * @type {Boolean}
-		 * @default false
-		 * @private
-		 */
-		pageScroll: PropTypes.bool,
-
-		/**
-		 * `true` if rtl, `false` if ltr.
-		 *
-		 * @type {Boolean}
-		 * @private
-		 */
-		rtl: PropTypes.bool,
-
-		/**
-		 * Spacing between items.
-		 *
-		 * @type {Number}
-		 * @default 0
-		 * @public
-		 */
-		spacing: PropTypes.number,
-
-		/**
-		 * Called to execute additional logic in a themed component when updating states and bounds.
-		 *
-		 * @type {Function}
-		 * @private
-		 */
-		updateStatesAndBounds: PropTypes.func
-	}
-
-	static defaultProps = {
-		cbScrollTo: nop,
-		data: [],
-		dataSize: 0,
-		direction: 'vertical',
-		overhang: 3,
-		pageScroll: false,
-		spacing: 0
-	}
-
-	constructor (props) {
-		super(props);
-
-		this.state = {firstIndex: 0, numOfItems: 0};
-	}
-
-	componentWillMount () {
-		if (this.props.clientSize) {
-			this.calculateMetrics(this.props);
-			this.updateStatesAndBounds(this.props);
+const VirtualListBaseFactory = (type) => {
+	return class VirtualListCore extends Component {
+		/* No displayName here. We set displayName to returned components of this factory function. */
+
+		static propTypes = /** @lends ui/VirtualList.VirtualListNative.prototype */ {
```

Comment:
`ui/VirtualList.VirtualListNative.prototype` seems not to be proper.

Reply:
Updated.